types/inquirer/inquirer-tests.ts

```diff
@@ -172,3 +177,37 @@ fetchAsyncQuestionProperty(
     paginator.paginate('test', 0);
     paginator.paginate('test', 0, 0);
 }
+
+{
+    const prompts = new Subject<DistinctQuestion>();
+    const promptResult = inquirer.prompt(prompts);
+    promptResult.ui.process.subscribe({
+        next: (value: {name: string, answer: any}) => {
+            // DO NOTHING
+        },
+    });
+    // $ExpectError
+    promptResult.ui.process.subscribe({
+        next: (value: {name_: string, answer: number}) => {
+            // DO NOTHING
+        },
+    });
+    prompts.complete();
+}
+
+{
+    const prompts = new Subject<DistinctQuestion<{str: string; num: number}>>();
+    const promptResult = inquirer.prompt(prompts);
+    promptResult.ui.process.subscribe({
+        next: (value: {name: 'str', answer: string} | {name: 'num', answer: number}) => {
+            // DO NOTHING
+        },
+    });
+    // $ExpectError
+    promptResult.ui.process.subscribe({
+        next: (value: {name: string, answer: number}) => {
+            // DO NOTHING
+        },
+    });
+    prompts.complete();
```

Comment:
Your current tests don't check whether the `QuestionAnswer` type is actually working.
Therefore I'd recommend you to add following test:
```suggestion

    promptResult.ui.process.subscribe({
        next: (value) => {
            if (value.name === "str") {
                // $ExpectType string
                value.answer;
            }
            else {
                // $ExpectType number
                value.answer;
            }
        },
    });

    prompts.complete();
```

Reply:
@manuth Thanks for reviewing, this is actually being tested in line 202 but another test will not hurt, will add it in a second

Reply:
Oooh fair - sorry, didn't catch that one